api/base/filters.py

```diff
@@ -365,11 +366,16 @@ def get_filtered_queryset(self, field_name, params, default_queryset):
         field = self.serializer_class._declared_fields[field_name]
         field_name = self.convert_key(field_name, field)
 
-        if isinstance(field, ser.SerializerMethodField):
+        if isinstance(field, ser.SerializerMethodField) and not isinstance(field, SerializerIntegerField):
             return_val = [
                 item for item in default_queryset
-                if self.FILTERS[params['op']](self.get_serializer_method(field_name)(item), params['value'])
+                if self.FILTERS[params['op']](self.get_serializer_method(field_name)(item), (params['value']))
             ]
+        elif isinstance(field, ser.SerializerMethodField) and isinstance(field, SerializerIntegerField):
```

Comment:
Minor, but you can take out the first clause and just do `elif isinstance(field, SerializerIntegerField):`